### ISSUE TITLE
RabbitMqPortal needs to be revisited because it's not functional in the current state

### ISSUE THREAD

**Issue by StefanOssendorf** [MarimerLLC/csla#3928]
The `RabbitMqPortal` implementation has two constructors.
https://github.com/MarimerLLC/csla/blob/a60f4da5d42538dcb24f5226a99f0f58c9bcbd61/Source/Csla.Channels.RabbitMq/RabbitMqPortal.cs#L33
https://github.com/MarimerLLC/csla/blob/a60f4da5d42538dcb24f5226a99f0f58c9bcbd61/Source/Csla.Channels.RabbitMq/RabbitMqPortal.cs#L58

As you can see we can either set the necessary services _or_ the data portal uri but _NOT_ both. Hence I dare to say this implementation doesn't work as inteded.
So we should revisit this implementation to make it right.

**Version and Platform**
CSLA version: Current

**Comment by rockfordlhotka** [comment]
Agreed. It should work with an options class like HttpProxy.

**Comment by rockfordlhotka** [comment]
I wonder if GrpcPortal has the same issue?

**Comment by StefanOssendorf** [comment]
> I wonder if GrpcPortal has the same issue?

I don't think so. Or at least I did not create an issue during adding the nullable stuff :D

**Comment by Freelancingonupwork** [comment]
Hello, i am not able to find any code related to this. May I know how I test it?

**Comment by rockfordlhotka** [comment]
> Hello, i am not able to find any code related to this. May I know how I test it?

It was originally something I created as a proof of concept, and so isn't well documented. I have now created a [readme](https://github.com/MarimerLLC/csla/blob/main/Source/Csla.Channels.RabbitMq/readme.md) describing the way it _should_ work.

The `RabbitMqPortal` type should really be getting its configuration from a `RabbitMqPortalOptions` instance (that type doesn't exist). And there should be a static `RabbitMqPortalServerExtensions` class that implements a `UseRabbitMqPortal` extension method that can be used on the host to configure the data portal host.
